### PR TITLE
Inf 226

### PR DESCRIPTION
**Pull request by dabrain34**

### CODE REVIEW

File: cerbero/utils/shell.py [hunk 1]
```diff
@@ -159,8 +159,17 @@ def unpack(filepath, output_dir):
         zf = zipfile.ZipFile(filepath, "r")
         zf.extractall(path=output_dir)
 
+def get_dl_mirror(url, mirror_url, filename = None):
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: if this function is private add _ before the function.
Missing documentation

File: cerbero/utils/shell.py [hunk 2]
```diff
@@ -159,8 +159,17 @@ def unpack(filepath, output_dir):
...
+      return url
+    if filename is None:
+      dl_mirror_url = mirror_url + os.path.basename(url)
+    else:
+      dl_mirror_url = mirror_url + filename
+    logging.info('Using the mirror url: %s instead of %s' % (dl_mirror_url, url))
+    return dl_mirror_url
 
-def download(url, destination=None, recursive=False, check_cert=True, user=None, password=None):
+def download(url, destination=None, recursive=False, check_cert=True, user=None, password=None, mirror_url=None, filename = None):
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: missing documentation for new parameters

File: cerbero/utils/shell.py [hunk 3]
```diff
@@ -173,6 +182,11 @@ def download(url, destination=None, recursive=False, check_cert=True, user=None,
     @param password: the password to use when connecting
     @type password: str
     '''
+    original_url =''
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: initialize to None

File: cerbero/utils/shell.py [hunk 4]
```diff
@@ -159,8 +159,17 @@ def unpack(filepath, output_dir):
...
 
+def get_dl_mirror(url, mirror_url, filename = None):
+    if mirror_url is None:
+      return url
+    if filename is None:
+      dl_mirror_url = mirror_url + os.path.basename(url)
+    else:
+      dl_mirror_url = mirror_url + filename
+    logging.info('Using the mirror url: %s instead of %s' % (dl_mirror_url, url))
+    return dl_mirror_url
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: pep8 requires 2 line breaks between methods

File: cerbero/utils/shell.py [hunk 5]
```diff
@@ -159,8 +159,17 @@ def unpack(filepath, output_dir):
...
         zf.extractall(path=output_dir)
 
+def get_dl_mirror(url, mirror_url, filename = None):
+    if mirror_url is None:
+      return url
+    if filename is None:
+      dl_mirror_url = mirror_url + os.path.basename(url)
+    else:
+      dl_mirror_url = mirror_url + filename
+    logging.info('Using the mirror url: %s instead of %s' % (dl_mirror_url, url))
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: Use cerbero's logging system

Comment (dabrain34) [reply cerbero/utils/shell.py]: I dont understand, the method above uses logger.info

Comment (ylatuya) [reply cerbero/utils/shell.py]: ok

File: cerbero/utils/shell.py [hunk 6]
```diff
@@ -160,7 +160,28 @@ def unpack(filepath, output_dir):
         zf.extractall(path=output_dir)
 
 
-def download(url, destination=None, recursive=False, check_cert=True, user=None, password=None):
+def _get_download_mirror_url(url, mirror_url, filename = None):
+    '''
+    Get the mirror url if available
+
+    @param url: original url to download
+    @type: str
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: @type url: str

File: cerbero/utils/shell.py [hunk 7]
```diff
@@ -160,7 +160,28 @@ def unpack(filepath, output_dir):
...
+    @param mirror_url: mirror url to use
+    @type destination: str
+    @param filename: custom filename to use
+    @type user: str
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: type file: str

File: cerbero/utils/shell.py [hunk 8]
```diff
@@ -172,7 +193,16 @@ def download(url, destination=None, recursive=False, check_cert=True, user=None,
     @type user: str
     @param password: the password to use when connecting
     @type password: str
+    @param mirror_url: mirror url to try first
+    @type: str
+    @param mirror_url: mirror url to try first
```
Comment (ylatuya) [inline cerbero/utils/shell.py]: duplicated

**Pull request merged**